juju/juju
# Remove controller cloud

## Description of change

Add support for removing a cloud from a controller. Broken into 3 commits:

1. Add state support for counting how many models a cloud has deployed and support for removing clouds with zero models.
2. Add upgrade steps to count and record the models for each cloud
3. Add API support for removing a cloud.

## QA steps

Nothing is wired up to use the remove yet. 
Deploy a 2.3 model and upgrade to 2.4 Check the cloud model counts.
Add/remove models and check cloud model counts.
